src/ds5/ds5-motion.cpp

```diff
@@ -451,46 +437,106 @@ namespace librealsense
                 calib_id = *reinterpret_cast<uint16_t*>(raw.data());
-                valid = true;
             }
             catch(const std::exception&)
             {
                 LOG_WARNING("IMU Calibration is not available, see the previous message");
             }
 
-            std::shared_ptr<mm_calib_parser> prs = nullptr;
             switch (calib_id)
             {
                 case ds::dm_v2_eeprom_id: // DM V2 id
-                    prs = std::make_shared<dm_v2_imu_calib_parser>(raw, _dev_cap, valid); break;
+                    prs = std::make_shared<dm_v2_imu_calib_parser>(raw, _dev_cap); break;
                 case ds::tm1_eeprom_id: // TM1 id
                     prs = std::make_shared<tm1_imu_calib_parser>(raw); break;
                 case ds::l500_eeprom_id: // L515
                     prs = std::make_shared<l500_imu_calib_parser>(raw); break;
                 default:
-                    throw recoverable_exception(to_string() << "Motion Intrinsics unresolved - "
-                                << ((valid)? "device is not calibrated" : "invalid calib type "),
-                                RS2_EXCEPTION_TYPE_BACKEND);
+                {
+                    // no valid calibration data at all, check other information to recover
+                    if (_l515_table_on_flash)
+                    {
+                        prs = std::make_shared<l500_imu_calib_parser>(raw);
+                    }
+                    else if ((bool)(_dev_cap & (ds::d400_caps::CAP_BMI_055) | ds::d400_caps::CAP_BMI_085))
+                    {
+                        prs = std::make_shared<dm_v2_imu_calib_parser>(raw, _dev_cap);
+                    }
+                    else
+                    {
+                        // should not come to this point as calibration table related issues should have handled
+                        throw recoverable_exception(to_string() << "Motion Intrinsics unresolved - "
+                            << "device is not calibrated - calibration data id: " << calib_id,
+                            RS2_EXCEPTION_TYPE_BACKEND);
+                    }
+                }
             }
             return prs;
         };
     }
 
+    std::vector<uint8_t> mm_calib_handler::get_imu_eeprom_raw() const
+    {
+        const int offset = 0;
+        const int size = ds::eeprom_imu_table_size;
+        command cmd(ds::MMER, offset, size);
+
+        std::vector<uint8_t> res;
+
+        // IMU table on D435i is in eeprom, less likely read issue, but in case return empty table
+        try
+        {
+            res = _hw_monitor->send(cmd);
+        }
+        catch (std::exception &e) {
+            LOG_WARNING(e.what());
+        }
+        catch (...) {}
+
+        return res;
+    }
+
+    std::vector<uint8_t> mm_calib_handler::get_imu_eeprom_raw_l515() const
+    {
+        // read imu calibration table on L515
+        // READ_TABLE 0x243 0
+        command cmd(0x43, 0x243, 0);
```

Comment:
defined READ_TABLE but kept the table id 0x243 as this is L515 table, if grouped with other D400 table ids, will confuse in the functions that uses the table ids.

in general, a future improvement, motion module and motion correction need some restucturing in the head files so that some of them will be common and some will be production line specific.